pkg/controller/mizar/mizar-service-controller.go

```diff
@@ -213,115 +214,143 @@ func (c *MizarServiceController) syncService(eventKeyWithType KeyWithEventType)
+	}
+
+	if tenantDefaultNetwork.Status.Phase != arktosapisv1.NetworkReady {
+		return errors.New(fmt.Sprintf("The arktos network %s is not Ready.", tenantDefaultNetwork.Name))
+	}
+
+	// update service spec with mizar annotation
+	obj, err := c.serviceLister.ServicesWithMultiTenancy(namespace, tenant).Get(name)
+	if err != nil {
+		return err
+	}
+	_, vpcNameOk := obj.Annotations[mizarAnnotationsVpcKey]
+	_, subnetNameOk := obj.Annotations[mizarAnnotationsSubnetKey]
 
-	if hasDNSServiceLabel && len(netName) != 0 {
-		klog.Infof("Starting ProcessArktosNetworkCreation network: %v", netName)
-		net, err := c.netClient.ArktosV1().NetworksWithMultiTenancy(service.Tenant).Get(netName, metav1.GetOptions{})
+	if !vpcNameOk && !subnetNameOk {
+		// assign default network when only there is no mizar annotation
+		// otherwise, this pod annotation needs to be fixed manually
+		if obj.Annotations == nil {
+			obj.Annotations = make(map[string]string)
+		}
+		obj.Annotations[mizarAnnotationsVpcKey] = getVPC(tenantDefaultNetwork)
+		obj.Annotations[mizarAnnotationsSubnetKey] = getSubnetNameFromVPC(tenantDefaultNetwork.Spec.VPCID)
+		_, err = c.kubeClientset.CoreV1().ServicesWithMultiTenancy(obj.Namespace, obj.Tenant).Update(obj)
+		klog.V(4).Infof("Add mizar annotation for service %s/%s/%s. error %v", obj.Tenant, obj.Namespace, obj.Name, err)
 		if err != nil {
-			klog.Errorf("The following network failed to get: %v", net)
-			return err
+			return errors.New(fmt.Sprintf("Service %s/%s/%s - update service's mizar annotation got error (%v)", obj.Tenant, obj.Namespace, obj.Name, err))
 		}
-		c.updateMizarVpcWithArktosName(net)
 	}
-	klog.Infof("Starting ProcessServiceCreation service: %v", service)
 
+	klog.V(4).Infof("Starting processServiceCreation service: %v", service)
+
+	// create service in mizar
 	msg := &BuiltinsServiceMessage{
 		Name:          service.Name,
-		ArktosNetwork: networkName,
 		Namespace:     service.Namespace,
 		Tenant:        service.Tenant,
 		Ip:            service.Spec.ClusterIP,
+		Vpc:           service.Annotations[mizarAnnotationsVpcKey],
+		Subnet:        service.Annotations[mizarAnnotationsSubnetKey],
 	}
 
 	response := c.grpcAdaptor.CreateService(c.grpcHost, msg)
 	code := response.Code
 	ip := response.Message
-	klog.Infof("Assigned ip by mizar is %v", ip)
+	klog.V(4).Infof("Assigned ip by mizar is %v", ip)
 
 	switch code {
 	case CodeType_OK:
-		klog.Infof("Mizar handled service creation successfully: %s", key)
-		if beginsWithKubernetes(service.Name) {
-			endpoint, err := c.kubeClientset.CoreV1().EndpointsWithMultiTenancy("default", "system").Get(kubernetesSvcDefaultName, metav1.GetOptions{})
+		klog.V(4).Infof("Mizar handled service creation successfully: %s", key)
+		if beginsWithKubernetes(service.Name) {	// Is this logic expected?
+			kubernetesEndpoint, err := c.kubeClientset.CoreV1().EndpointsWithMultiTenancy(metav1.NamespaceDefault, metav1.TenantSystem).Get(kubernetesSvcDefaultName, metav1.GetOptions{})
```

Comment:
a service can have multiple endpoints. so this should be xxxEndpoints.

also, let's use arktos instead of kubernetes

Reply:
This part is old that I did not fully understand. Will need further debugging to understand better. The service name started with kubernetes, I don't think it is the branding issue that needs to be changed now. If we need to change, all service name should be updated to arktos.